dtos/matches/query-match.dto.ts

```diff
@@ -1,33 +1,8 @@
-import { ApiPropertyOptional } from '@nestjs/swagger';
-import { IsNumberString, IsOptional, IsEnum } from 'class-validator';
-import { MatchStatusType } from './match.dto';
+import { MatchDto } from './match.dto';
+import { PickType } from '@nestjs/swagger';
 
-export class QueryMatchDto {
-  /*@ApiPropertyOptional({
-    description: 'The id of a match',
-  })
-  @IsPositive()
-  @IsInt()
-  id: number;
-*/
-  @ApiPropertyOptional({
-    description: "The id of the 'home' user",
-  })
-  @IsOptional()
-  @IsNumberString()
-  homeId?: string;
-
-  @ApiPropertyOptional({
-    description: "The id of the 'away' user",
-  })
-  @IsOptional()
-  @IsNumberString()
-  awayId?: string;
-
-  @ApiPropertyOptional({
-    description: 'The outcome of the match',
-  })
-  @IsOptional()
-  @IsEnum(MatchStatusType)
-  status?: MatchStatusType;
-}
+export class QueryMatchDto extends PickType(MatchDto, [
```

Comment:
```suggestion
export class QueryMatchDto extends PickType(PartialType(MatchDto), [
```
These properties are optional in the query, required in the original dto. Also note that it may break functionality because `homeId` and `awayId` are now `number`s.

Reply:
Are you sure they're not optional iin the original too ? Anyway, I think the type issue explains why you re-defined  it actually, and there is no purpose in using a mapped type I think, I'll just revert to what you did and only change the main dto's name to avoid confusion.